smqtk_descriptors/impls/descriptor_generator/pytorch.py

```diff
@@ -0,0 +1,538 @@
+        None then implementations should attempt to use their "pre-trained"
+        mode if they have one. Otherwise an error is raised on model load when
+        not provided.
+    :param int|None image_tform_threads:
+        Number of threads to utilize when transforming image matrices for
+        network input. Set to `None` to use all available CPU threads.
+        This is 1 (serial) by default).
+    :param int batch_size:
+        Batch size use for model computation.
+    :param bool use_gpu:
+        If the model should be loaded onto, and processed on, the GPU.
+    :param None|int cuda_device:
+        Specific device value to pass to the CUDA transfer calls if `use_gpu`
+        us enabled.
+    :param None|int|float|str normalize:
+        Optionally normalize descriptor vectors as they are produced. We use
+        ``numpy.linalg.norm`` so any valid value for the ``ord`` parameter is
+        acceptable here.
+    :param bool iter_runtime:
+        By default the input image matrix data is accumulated into a list in
+        order to utilize `torch.utils.data.DataLoader` for batching.
+        If this is parameter is True however, we use a custom parallel
+        iteration pipeline for image transformation into network appropriate
+        tensors.
+        Using this mode changes the maximum RAM usage profile to be linear with
+        batch-size instead of input image data amount, as well as having lower
+        latency to first yield.
+        This mode is idea for streaming input or high input volume situations.
+        This mode currently has a short-coming of working only in a threaded
+        manner so it is not as fast as using the `DataLoader` avenue.
+    :param bool global_average_pool:
+        Optionally apply a GAP operation to the spatial dimension of the feature
+        vector that is returned by the descriptor generator. Some models return
+        a w x h x k tensor and flatten them into a single dimension, which can
+        remove the spatial context of the features. Taking an average over each
+        channel can improve the clustering in some cases.
+    """
+
+    @classmethod
+    def is_usable(cls) -> bool:
+        valid = torch is not None
+        if not valid:
+            LOG.debug("Torch python module not imported")
+        return valid
+
+    @classmethod
+    def get_default_config(cls) -> Dict[str, Any]:
+        c = super().get_default_config()
+        c['image_reader'] = make_default_config(ImageReader.get_impls())
+        return c
+
+    @classmethod
+    def from_config(
+        cls: Type[T],
+        config_dict: Dict,
+        merge_default: bool = True
+    ) -> T:
+        # Copy config to prevent input modification
+        config_dict = copy.deepcopy(config_dict)
+        config_dict['image_reader'] = \
+            from_config_dict(config_dict['image_reader'],
+                             ImageReader.get_impls())
```

Comment:
In an effort to reduce the use of `\`s, we can reformat this to
```suggestion
        config_dict['image_reader'] = from_config_dict(
            config_dict['image_reader'],
            ImageReader.get_impls()
        )
```